examples/media/multipart/src/main/java/io/helidon/examples/media/multipart/FileService.java

```diff
@@ -88,72 +88,45 @@ private void upload(ServerRequest req, ServerResponse res) {
     }
 
     private void bufferedUpload(ServerRequest req, ServerResponse res) {
-        req.content().as(ReadableMultiPart.class).thenAccept(multiPart -> {
-            for (ReadableBodyPart part : multiPart.fields("file[]")) {
-                writeBytes(storage.create(part.filename()), part.as(byte[].class));
-            }
-            res.status(Http.Status.MOVED_PERMANENTLY_301);
-            res.headers().put(Http.Header.LOCATION, "/ui");
-            res.send();
-        });
+        req.content().as(ReadableMultiPart.class)
+                .flatMapIterable(multiPart -> multiPart.fields("file[]"))
+                .flatMap(part -> part.content()
+                        .map(DataChunk::data)
+                        .flatMapIterable(Arrays::asList)
+                        .to(IoMulti.writeToFile(storage.create(part.filename()))
+                                .executor(executor)
+                                .build())
+                )
+                .onError(throwable -> {
+                    res.status(Http.Status.INTERNAL_SERVER_ERROR_500);
+                    res.send(throwable.toString());
+                })
+                .onComplete(() -> {
+                    res.status(Http.Status.MOVED_PERMANENTLY_301);
+                    res.headers().put(Http.Header.LOCATION, "/ui");
+                    res.send();
+                })
+                .ignore();
```

Comment:
It look much better with @tomas-langer 's proposed `IoMulti.write..` abstraction